lms/static/js/edxnotes/plugins/accessibility.js

```diff
@@ -0,0 +1,253 @@
+
+            // Note and ddit, delete and close buttons always come in quadruplets
+            for (i = 0; i < editButtons.length; i++) {
+                tabControls.push($(notes.get(i)));
+                tabControls.push($(editButtons.get(i)));
+                tabControls.push($(delButtons.get(i)));
+                tabControls.push($(closeButtons.get(i)));
+            }
+
+            return tabControls;
+        },
+
+        getEditorTabControls: function () {
+            var editor, editorControls, textArea, save, cancel, tabControls = [];
+
+            // Editor elements
+            editor = this.annotator.element.find('.annotator-editor');
+            editorControls = editor.find('.annotator-controls');
+            textArea = editor.find('.annotator-listing')
+                             .find('.annotator-item')
+                             .first()
+                             .children('textarea');
+            save  = editorControls.find('.annotator-save');
+            cancel = editorControls.find('.annotator-cancel');
+
+            tabControls.push($(textArea.get(0)), $(save.get(0)), $(cancel.get(0)));
+
+            return tabControls;
+        },
+
+        focusOnNextTabControl: function (tabControls, tabControl) {
+            var nextIndex;
+
+            _.each(tabControls, function (element, index) {
+                if (element.is(tabControl)) {
+                    nextIndex = index === tabControls.length - 1 ? 0 : index + 1;
+                    tabControls[nextIndex].focus();
+                }
+
+            });
+        },
+
+        focusOnPreviousTabControl: function (tabControls, tabControl) {
+            var previousIndex;
+            _.each(tabControls, function (element, index) {
+                if (element.is(tabControl)) {
+                    previousIndex = index === 0  ? tabControls.length - 1 : index - 1;
+                    tabControls[previousIndex].focus();
+                }
+            });
+        },
+
+        onHighlightKeyDown: function (event) {
+            var KEY = $.ui.keyCode,
+                keyCode = event.keyCode,
+                target = $(event.currentTarget),
+                annotations, position,
+                self = this;
+
+            switch (keyCode) {
+                case KEY.ENTER:
+                case KEY.SPACE:
+                    if (!this.annotator.viewer.isShown()) {
+                        position = target.position();
+                        annotations = target.parents('.annotator-hl').addBack().map(function() {
+                            return $(this).data('annotation');
+                        });
+                        this.annotator.showViewer($.makeArray(annotations), {top: position.top, left: position.left});
+                        setTimeout(function() {
+                            // Focus on listing
+                            self.annotator.element.find('.annotator-listing').focus();
```

Comment:
@polesye 
> Focusing on this element will force to read the entire content of the dialog (even with button labels). Is it expected behavior?

Actually, nothing is read at all with VoiceOver when the `<ul>` gets the focus programmatically, even if an `aria-label = "Viewer window"` is added.

@cptvitamin Any suggestion there? Or more generally, what ARIA attributes have to be added to the two modal windows, Viewer and Editor in order for them to have something read when they get the focus programmatically? The Viewer is a `<ul>` and the Editor a `<form>`.